TestFluxEstimator.test_with_power_law fail in master

@joleroi - I think after your changes yesterday `TestFluxEstimator.test_with_power_law` fails in master. Looks like the energy grouping isn't applied to `backscale_ratio`?

```
E       TypeError: statistic input array sizes do not match
```

https://travis-ci.org/gammapy/gammapy/jobs/162223291#L938

I will mark that test as `xfail` for now.

Putting this under the 0.5 milestone.
Let me know if you don't have time, and I'll try to do it on Monday.

--- a/gammapy/spectrum/tests/test_flux_point.py
+++ b/gammapy/spectrum/tests/test_flux_point.py
@@ -257,7 +257,6 @@ class TestFluxEstimator:
 
     # TODO: fix this test
     # see https://github.com/gammapy/gammapy/issues/724
-    @pytest.mark.xfail
     def test_with_power_law(self):
         # import logging
         # logging.basicConfig(level=logging.DEBUG)

--- a/gammapy/spectrum/tests/test_flux_point.py
+++ b/gammapy/spectrum/tests/test_flux_point.py
@@ -255,8 +255,6 @@ class TestFluxEstimator:
 
         self.groups = self.seg.groups
 
-    # TODO: fix this test
-    # see https://github.com/gammapy/gammapy/issues/724
     def test_with_power_law(self):
         # import logging
         # logging.basicConfig(level=logging.DEBUG)

--- a/gammapy/stats/tests/test_fit_statistics.py
+++ b/gammapy/stats/tests/test_fit_statistics.py
@@ -50,9 +50,7 @@ def test_cash():
     assert_allclose(actual, desired)
 
 
-# TODO: fix this test
-# https://github.com/gammapy/gammapy/issues/724#issuecomment-249362709
-@pytest.mark.xfail
+# TODO: Make this test independent of Sherpa
 @requires_dependency('sherpa')
 def test_wstat():
     import sherpa.stats as ss

--- a/gammapy/stats/tests/test_fit_statistics.py
+++ b/gammapy/stats/tests/test_fit_statistics.py
@@ -50,7 +50,6 @@ def test_cash():
     assert_allclose(actual, desired)
 
 
-# TODO: Make this test independent of Sherpa
 @requires_dependency('sherpa')
 def test_wstat():
     import sherpa.stats as ss
@@ -77,18 +76,17 @@ def test_wstat():
                        exposure_time=[1, 1],
                        backscale_ratio=1. / alpha[test_bin],
                        data_size=1)
-
     desired_testbin, fvec = sherpa_stat.calc_stat(data[test_bin],
                                                   model[test_bin],
                                                   staterror=staterror[test_bin],
-                                                  bkg=bkg_testbin)
+                                                  extra_args=bkg_testbin)
 
     actual_testbin = statsvec[test_bin]
     #    assert_allclose(actual_testbin, desired_testbin)
 
     # Now check total stat for all bins
     desired, fvec = sherpa_stat.calc_stat(data, model, staterror=staterror,
-                                          bkg=bkg)
+                                          extra_args=bkg)
 
     actual = np.sum(statsvec)
     print(fvec)

